File: android/src/main/kotlin/cl/ceisufro/fluttermbtilesextractor/ExtractRequest.kt
package cl.ceisufro.fluttermbtilesextractor


class ExtractRequest(val pathToDB: String,
                     val desiredPath: String,
                     val requestPermissions: Boolean,
                     val removeAfterExtract: Boolean,
                     val stopOnError: Boolean,
                     val onlyReference: Boolean,
                     val returnReference: Boolean,
                     val schema: Int) {
    companion object {
        fun fromMap(map: Map<String, Any>): ExtractRequest {
            val pathToDB = map["pathToDB"] as String
            val desiredPath = map["desiredPath"] as String
            val requestPermissions = map["requestPermissions"] as Boolean
            val removeAfterExtract = map["removeAfterExtract"] as Boolean
            val stopOnError = map["stopOnError"] as Boolean
            val onlyReference = map["onlyReference"] as Boolean
            val returnReference = map["returnReference"] as Boolean
            val schema = map["schema"] as Int
            return ExtractRequest(pathToDB, desiredPath, requestPermissions, removeAfterExtract, stopOnError, onlyReference, returnReference, schema)
        }
    }

    fun toMap(): Map<String, Any> {
        val map = hashMapOf<String, Any>()
        map.put("pathToDB", pathToDB)
        map.put("desiredPath", desiredPath)
        map.put("requestPermissions", requestPermissions)
        map.put("removeAfterExtract", removeAfterExtract)
        map.put("stopOnError", stopOnError)
        map.put("onlyReference", onlyReference)
        map.put("returnReference", returnReference)
        map.put("schema", schema)
        return map
    }
}

File: ios/Classes/ExtractRequest.swift
//
//  ExtractRequest.swift
//  flutter_mbtiles_extractor
//
//  Created by CEISUFRO on 09-07-18.
//

import Foundation
class ExtractRequest:NSObject{
    let pathToDB:String
    let desiredPath:String
    let requestPermissions:Bool
    let removeAfterExtract:Bool
    let stopOnError:Bool
    let onlyReference:Bool
    let returnReference:Bool
    let schema:Int

    init(pathToDB:String,desiredPath:String,requestPermissions:Bool,removeAfterExtract:Bool,stopOnError:Bool,onlyReference:Bool,returnReference:Bool,schema:Int) {
        self.pathToDB = pathToDB
        self.desiredPath=desiredPath
        self.requestPermissions=requestPermissions
        self.removeAfterExtract=removeAfterExtract
        self.stopOnError=stopOnError
        self.onlyReference=onlyReference
        self.returnReference=returnReference
        self.schema=schema
    }
    
    static func fromMap(map:NSDictionary) -> ExtractRequest{
        let pathToDB:String = map["pathToDB"] as! String
        let desiredPath:String = map["desiredPath"] as! String
        let requestPermissions:Bool = map["requestPermissions"] as! Bool
        let removeAfterExtract:Bool = map["removeAfterExtract"] as! Bool
        let stopOnError:Bool = map["stopOnError"] as! Bool
        let onlyReference:Bool = map["onlyReference"] as! Bool
        let returnReference:Bool = map["returnReference"] as! Bool
        let schema:Int = map["schema"] as! Int
        return ExtractRequest(
            pathToDB: pathToDB,
            desiredPath: desiredPath,
            requestPermissions: requestPermissions,
            removeAfterExtract: removeAfterExtract,
            stopOnError: stopOnError,
            onlyReference: onlyReference,
            returnReference: returnReference,
            schema: schema)
    }
    
    func toMap() -> NSDictionary{
        return [
            "pathToDB":self.pathToDB,
            "desiredPath":self.desiredPath,
            "requestPermissions":self.requestPermissions,
            "removeAfterExtract":self.removeAfterExtract,
            "stopOnError":self.stopOnError,
            "onlyReference":self.onlyReference,
            "returnReference":self.returnReference,
            "schema":self.schema,
        ]
    }
}
